### ISSUE TITLE
iOS 12 OpenVPN Client Issue with sending traffic

### ISSUE THREAD

**Issue by troyfontaine** [pivpn/pivpn#623]
## Issue
I'm not sure if this is configuration related or not?  The OpenVPN configuration works fine on my Macbook Pro on Mojave using Tunnelblick, but the OpenVPN configuration doesn't seem to function on my iPhone running iOS 12 with OpenVPN Connect.  What does work from my phone?  DNS resolution and port scanning.  I've tested it from my Macbook and I can SSH and surf the internet.

Scenario: PiVPN on a Pi 3b with static IP on Raspbian Stretch.  It is behind an Ubiquiti USG 3P Router with TCP 443 port forwarded.  This has worked previously.

There is an odd entry in the log output from OpenVPN Connect where it appears that it resolves my DNS name to an IPv6 address-but the address does not correspond to my external IPv6 IP.  I also don't have my dynamic DNS configured to set up a AAAA record.

```
2018-10-10 18:18:08 EVENT: CONNECTED mydomain.net:443 (2604:5580:21:::) via /TCPv6 on NetworkExtensionTUN/10.8.0.4/ gw=[/]
```

If I should open this as an issue elsewhere, please let me know.

## Have you searched for similar issues and solutions? 
Yes, but nothing seems to match?

## Console output of `pivpn debug`
```
commit de76940b65bab86eda772158da58328482363559
Merge: 18bfa92 58fcaaf
Author: redfast00 <redfast00@users.noreply.github.com>
Date:   Wed Aug 15 21:17:33 2018 +0200

    Merge pull request #554 from Giraffe1966/issue-552

    Fix issue #552 (remove dead code)
:::					:::
::	Recursive list of files in	 ::
::	/etc/openvpn/easy-rsa/pki	 ::
:::					:::
/etc/openvpn/easy-rsa/pki/:
Default.txt
ca.crt
crl.pem
index.txt
index.txt.attr
index.txt.attr.old
index.txt.old
issued
private
serial
serial.old
ta.key

/etc/openvpn/easy-rsa/pki/issued:
server_knkYu01HfdUetI1J.crt

/etc/openvpn/easy-rsa/pki/private:
ca.key
server_knkYu01HfdUetI1J.key

:::					:::
::	Output of /etc/pivpn/*		 ::
:::					:::
:: START /etc/pivpn/DET_PLATFORM ::
Raspbian
:: END /etc/pivpn/DET_PLATFORM ::
:: START /etc/pivpn/INSTALL_PORT ::
443
:: END /etc/pivpn/INSTALL_PORT ::
:: START /etc/pivpn/INSTALL_PROTO ::
tcp
:: END /etc/pivpn/INSTALL_PROTO ::
:: START /etc/pivpn/INSTALL_USER ::
pi
:: END /etc/pivpn/INSTALL_USER ::
:: START /etc/pivpn/NO_UFW ::
0
:: END /etc/pivpn/NO_UFW ::
:: START /etc/pivpn/pivpnINTERFACE ::
eth0
:: END /etc/pivpn/pivpnINTERFACE ::
:: START /etc/pivpn/setupVars.conf ::
pivpnUser=pi
UNATTUPG=unattended-upgrades
pivpnInterface=eth0
IPv4dns=192.168.0.1
IPv4addr=192.168.0.90
IPv4gw=192.168.0.1
pivpnProto=tcp
PORT=443
ENCRYPT=2048
APPLY_TWO_POINT_FOUR=true
DOWNLOAD_DH_PARAM=
PUBLICDNS=mydomain.net
OVPNDNS1=192.168.0.250
OVPNDNS2=
:: END /etc/pivpn/setupVars.conf ::
:: START /etc/pivpn/setupVars.conf.update.bak ::
pivpnUser=pi
UNATTUPG=unattended-upgrades
pivpnInterface=eth0
IPv4dns=192.168.0.1
IPv4addr=192.168.0.90
IPv4gw=192.168.0.1
pivpnProto=tcp
PORT=443
ENCRYPT=2048
APPLY_TWO_POINT_FOUR=true
DOWNLOAD_DH_PARAM=
PUBLICDNS=mydomain.net
OVPNDNS1=192.168.0.250
OVPNDNS2=
:: END /etc/pivpn/setupVars.conf.update.bak ::
:: START /etc/pivpn/TWO_POINT_FOUR ::
:: END /etc/pivpn/TWO_POINT_FOUR ::
:::					:::
:: /etc/openvpn/easy-rsa/pki/Default.txt ::
:::					:::
client
dev tun
proto tcp
remote mydomain.net 443
resolv-retry infinite
nobind
persist-key
persist-tun
remote-cert-tls server
tls-version-min 1.2
verify-x509-name server_knkYu01HfdUetI1J name
cipher AES-256-CBC
auth SHA256
compress lz4
verb 3
```

## Console Output of `sudo iptables -t nat -S`
```
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-A POSTROUTING -s 10.8.0.0/24 -o eth0 -j MASQUERADE
```

## Console Output of `sudo iptables -S`
```
-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT
```

## output of `sudo netstat -uanp | grep openvpn`

```
no output available
```

## Have you taken any steps towards solving your issue? 
I have removed and reinstalled pivpn, re-imaged my Pi3 and re-installed, attempted modifying the configuration adding the bypass-dhcp option to the default gateway.  I'm kicking myself for wiping out my previously working install to switch Pi OS distros.

**Comment by troyfontaine** [comment]
Okay, after further digging this could be related to the use of elliptic curve crypto:
```ecdh-curve secp384r1``` I’ll try and swap it out.

**Comment by troyfontaine** [comment]
And it isn't an issue with PiVPN or OpenVPN-it is the OpenVPN Connect client based on the ticket [here](https://community.openvpn.net/openvpn/ticket/1107).

**Comment by troyfontaine** [comment]
Looks like it was related to the `compress` configuration.  The iOS client warns about using it (there was a security vulnerability announced in August).  If you have it disabled client side and enabled server side things don’t work.

So that configuration should probably be removed-or a warning added that users of OpenVPN connect should enable downstream only or full for compression if the development team desires to keep the `compress lz4` configuration in.

I didn’t see any performance impact with speedtest.net tests with that configuration change.

**Comment by troyfontaine** [comment]
#597 would be great to see merged!

**Comment by troyfontaine** [comment]
Solved by #597
